StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentSheetError.swift

```diff
@@ -15,12 +15,40 @@ import StripePayments
 /// Most errors do not originate from PaymentSheet itself; instead, they come from the Stripe API
 /// or other SDK components like STPPaymentHandler, PassKit (Apple Pay), etc.
 public enum PaymentSheetError: Error {
-    /// An unknown error.
-    case unknown(debugDescription: String)
 
+    // MARK: Generic errors
+    case missingClientSecret
+    case invalidClientSecret
+    case unexpectedResponseFromStripeAPI
+    case applePayNotSupported
+    case alreadyPresented
+    case flowControllerConfirmFailed(message: String)
+    case errorHandlingNextAction
+    case unrecognizedHandlerStatus
+    case accountLinkFailure
     /// No payment method types available error.
     case noPaymentMethodTypesAvailable(intentPaymentMethods: [STPPaymentMethodType])
 
+    // MARK: Loading errors
+    case paymentIntentInTerminalState(status: STPPaymentIntentStatus)
+    case setupIntentInTerminalState(status: STPSetupIntentStatus)
+    case fetchPaymentMethodsFailure
+
+    // MARK: Deferred intent errors
+    case deferredIntentValidationFailed(message: String)
```

Comment:
I left this and `flowControllerConfirmFailed` to have a free form String messages for convince. Given the narrow scope of these errors I don't think they will be abused to accidentally send PII in the error message, which is something I believe .unknown was susceptible to.